pkg/auth/authmap_gc.go

```diff
@@ -405,30 +408,46 @@ func (r *authMapGarbageCollector) cleanupExpiredEntries(_ context.Context) error
 
 func (r *authMapGarbageCollector) subscribeToEndpointEvents(endpointManager endpointmanager.EndpointManager) {
 	endpointManager.Subscribe(r)
+
+	r.endpointsCacheMutex.Lock()
+	r.endpointsCache = map[uint16]*endpoint.Endpoint{}
+	for _, ep := range endpointManager.GetEndpoints() {
+		r.endpointsCache[ep.GetID16()] = ep
+	}
+	r.endpointsCacheSynced = true
+	r.endpointsCacheMutex.Unlock()
 }
 
 func (r *authMapGarbageCollector) EndpointCreated(ep *endpoint.Endpoint) {
-	// noop function to satisfy interface
+	r.endpointsCacheMutex.Lock()
+	r.endpointsCache[ep.GetID16()] = ep
+	r.endpointsCacheMutex.Unlock()
 }
 
 func (r *authMapGarbageCollector) EndpointDeleted(ep *endpoint.Endpoint, conf endpoint.DeleteConfig) {
+	r.endpointsCacheMutex.Lock()
+	delete(r.endpointsCache, ep.GetID16())
+	r.endpointsCacheMutex.Unlock()
+
 	// when an endpoint got removed clean the authmap entries
```

Comment:
Why do we cleanup endpoints immediately after receiving the deletion event? the rest of the authmap garbage collector  works the way that all events are captured first - and the actual GC execution removes them. 

The reason why I'm asking is because this also includes proper reporting of the error back to the timer-based job of the Hive-framework. Instead, only a warning gets logged here.

IMO we should fix and align this - even though this might be out of scope for this PR because it's not related to the actual issue and has been introduced with the EP garbage collection.

But i'd like to see this in a follow up PR if possible.

Reply:
The reason for this is that this change was done in hardening against possible attacks of stale entries in the auth map. Acting on deletion event itself would close the time gap where this is possible.

The error handling aspect of this might be a good follow up, feel free to file an issue for it happy to look into it